examples/expressions/10_expressions_intro_new.py

```diff
@@ -0,0 +1,201 @@
+# `current_annual_salary` is the target variable we want to predict.
+#
+
+from skrub import TableReport
+
+TableReport(training_data)
+# %%
+# Our goal is to predict the `current_annual_salary` of employees based on their
+# other features. We will use skrub's DataOps to combine both skrub and scikit-learn
+# objects into a single DataOps plan, which will allow us to preprocess the data,
+# train a model, and tune hyperparameters.
+#
+# We begin by defining a skrub variable, which is the entry point for our DataOps plan.
+# Additionally, we use the subsample function to limit the number of rows that
+# are used to produce previews of each step in the plan. This is useful to speed up
+# the development process. At training time, the DataOps will automatically
+# use the full dataset.
+
+import skrub
+
+data_var = skrub.var("data", training_data).skb.subsample()
+
+# %%
+# Next, we define the starting feature matrix ``X`` and the target variable ``y``.
+# We will use the `skb.mark_as_X` and `skb.mark_as_y` methods to mark these variables
+# in the DataOps plan. This allows skrub to properly split these objects into
+# training and validation steps when executing cross-validation or hyperparameter
+# tuning.
+
+X = data_var.drop("current_annual_salary", axis=1).skb.mark_as_X()
+y = data_var["current_annual_salary"].skb.mark_as_y()
+# %%
+# Our first step is to vectorize the features in ``X``. We will use the
+# `TableVectorizer` to convert the categorical and numerical features into a
+# numerical format that can be used by machine learning algorithms.
+# We apply the vectorizer to ``X`` using the `skb.apply` method, which allows us to
+# apply any scikit-learn compatible transformer to the skrub variable.
+
+from skrub import TableVectorizer
+
+vectorizer = TableVectorizer()
+
+X_vec = X.skb.apply(vectorizer)
+X_vec
+# %%
+# By clicking on ``Show graph``, we can see the DataOps plan that has been created:
+# the plan shows the steps that have been applied to the data so far.
+# Now that we have the vectorized features, we can proceed to train a model.
+# We use a scikit-learn `HistGradientBoostingRegressor` to predict the target variable.
+# We apply the model to the vectorized features using ``.skb.apply``, and pass
+# ``y`` as the target variable.
+# Note that the resulting ``predictor`` will show the prediction results on the
+# preview subsample, but the actual model has not been fitted yet.
+
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+hgb = HistGradientBoostingRegressor()
+
+predictor = X_vec.skb.apply(hgb, y=y)
+predictor
+
+# %%
+# Now that we have built our entire plan, we can have explore it in more detail
+# with the ``.skb.full_report()`` method. This method evaluates each step in
+# the plan and shows detailed information about the operations that are being performed.
+# The full plan is shown in a browser window and saved on disk.
+predictor.skb.full_report()
```

Comment:
The report does not appear in the render HTML on the website. We need to find a way for the reader to see this, elsewhere it is of limited use and is confusing